pkg/operator/controller/ingress/deployment.go

```diff
@@ -366,7 +366,11 @@ func desiredRouterDeployment(ci *operatorv1.IngressController, ingressController
 		env = append(env, corev1.EnvVar{Name: "ROUTER_USE_PROXY_PROTOCOL", Value: "true"})
 	}
 
-	env = append(env, corev1.EnvVar{Name: "ROUTER_THREADS", Value: "4"})
+	threads := 4
+	if ci.Spec.Threading.Count > 0 {
+		threads = ci.Spec.Threading.Count
+	}
+	env = append(env, corev1.EnvVar{Name: "ROUTER_THREADS", Value: strconv.Itoa(threads)})
```

Comment:
Could you add unit test cases to cover this code? Should be pretty straightforward. See https://github.com/openshift/cluster-ingress-operator/pull/576 for an example.

Reply:
I added some checks for `ROUTER_THREADS` to the existing `TestDesiredRouterDeployment` test case to cover this